cmd/zed/manage/compact.go

```diff
@@ -0,0 +1,86 @@
+package manage
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/brimdata/zed/cli/commitflags"
+	"github.com/brimdata/zed/cli/lakeflags"
+	apicompact "github.com/brimdata/zed/cmd/zed/manage/compact"
+	"github.com/brimdata/zed/lake/api"
+	"github.com/brimdata/zed/pkg/charm"
+	"gopkg.in/yaml.v3"
+)
+
+var compact = &charm.Spec{
+	Name:  "compact",
+	Usage: "compact",
+	Short: "compact objects in a pool",
+	New:   newCompact,
+}
+
+type compactCommand struct {
+	*Command
+	commitFlags commitflags.Flags
+	coldthresh  time.Duration
+	config      Config
+}
+
+func newCompact(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
+	c := &compactCommand{Command: parent.(*Command)}
+	c.commitFlags.SetFlags(f)
+	f.Func("config", "path of manage yaml config file", func(s string) error {
+		b, err := os.ReadFile(s)
+		if err != nil {
+			return err
+		}
+		return yaml.Unmarshal(b, &c.config)
+	})
+	f.DurationVar(&c.config.ColdThreshold, "coldthresh", time.Minute*5, "age at which objects are considered for compaction")
```

Comment:
Any reason not to put this in the configuration file?

Reply:
You mean just have it exclusively set in the config file? It's nice to be able to override config file options via command line variable, but I'm cool with just having the config file for now.